pkg/services/validator/maas.go

```diff
@@ -286,7 +266,11 @@ func readMaasResource() (vpmaasapi.Resource, error) {
 	}
 	res.Disk = disk
 
-	pool, err := prompts.ReadText("Machine pool", "", true, -1)
+	resourcePools, err := clouds.GetMaasResourcePools(c)
+	if err != nil {
+		return res, err
+	}
+	pool, err := prompts.Select("Machine pool", resourcePools)
```

Comment:
nit: "Resource pool"

Reply:
Also, to make this optional, you'll have to insert an "N/A" option into the selection.